### PR TITLE
Feature/ecom 2204  php client create new endpoint add order status by payment route

### PR DESCRIPTION
**Pull request by Francois-Gomis**
### Reason for change

https://linear.app/almapay/issue/ECOM-2204/[-php-client]-create-new-endpoint
### Code changes

New endpoint in payment for order status
### How to test

_As a reviewer, you are encouraged to test the PR locally._

### Checklist for authors and reviewers

- [ ] The title of the PR uses business wording, not technical jargon, for the changelog readers to understand it
- [ ] The PR implements the changes asked in the referenced task / issue
- [ ] The automated tests are compliant with the [testing strategy](https://www.notion.so/almapay/Backend-testing-strategy-06c642cec1bf47b9b8feca3a91ea8d4a)
- [ ] The tests are relevant, and cover the corner/error cases, not only the happy path
- [ ] You understand the impact of this PR on existing code/features
- [ ] The changes include adequate logging and Datadog traces
- [ ] Documentation is updated (API, developer documentation, ADR, Notion...)

### CODE REVIEW

File: src/Endpoints/Payments.php [hunk 1]
```diff
@@ -355,4 +388,28 @@ public function sendSms($id)
...
 
+    /**
+     * @param $paymentId
+     * @param $merchantOrderReference
+     * @param $status
+     * @param $isShipped
+     * @return void
+     * @throws ParametersException
+     */
+    public function checkAddOrderStatusParams($paymentId, $merchantOrderReference, $status, $isShipped)
```
Comment (webaaz) [inline src/Endpoints/Payments.php:399]: non need to be public

File: src/Endpoints/Payments.php [hunk 2]
```diff
@@ -335,6 +337,37 @@ public function addOrder($id, $orderData, $overwrite = false)
...
+     * @throws RequestError
+     * @throws RequestException
+     */
+    public function addOrderStatusByMerchantOrderReference(
+        $paymentId,
+        $merchantOrderReference,
+        $status,
+        $isShipped = null)
+    {
+        $this->checkAddOrderStatusParams($paymentId, $merchantOrderReference, $status, $isShipped);
```
Comment (webaaz) [inline src/Endpoints/Payments.php:358]: Not fan to check 4 params in only one call:
-> one call per param
or
-> test here

File: src/Lib/BoolUtils.php [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    /**
+     * @param $boolToCheck
+     * @return bool
+     */
+    public static function isStrictlyBoolOrNull($boolToCheck)
+    {
+        return is_bool($boolToCheck) || $boolToCheck === null;
+    }
+
+}
```
Comment (webaaz) [inline src/Lib/BoolUtils.php:16]: Need on blank line at the end of file

File: src/Lib/StringUtils.php [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    /**
+     * @param $stringToCheck
+     * @return bool
+     */
+    public static function isAValidString($stringToCheck)
+    {
+        return is_string($stringToCheck) && trim($stringToCheck) !== '';
+    }
+
+}
```
Comment (webaaz) [inline src/Lib/StringUtils.php:16]: Need on blank line at the end of file

File: src/Endpoints/Payments.php [hunk 5]
```diff
@@ -335,6 +337,37 @@ public function addOrder($id, $orderData, $overwrite = false)
...
+     * @return void
+     * @throws ParametersException
+     * @throws RequestError
+     * @throws RequestException
+     */
+    public function addOrderStatusByMerchantOrderReference(
+        $paymentId,
+        $merchantOrderReference,
+        $status,
+        $isShipped = null)
```
Comment (webaaz) [inline src/Endpoints/Payments.php:356]: parentheses on a new line

File: tests/Unit/Endpoints/PaymentsTest.php [hunk 6]
```diff
@@ -451,4 +452,152 @@ public function testFullRefundRequestError()
...
+        $this->expectException(ParametersException::class);
+        $paymentEndpoint->addOrderStatusByMerchantOrderReference(
+            $paymentId,
+            $merchantOrderReference,
+            $status,
+            $isShipped
+        );
+    }
+
+    public function testAddIOrderStatusThrowRequestExceptionForNon200Return()
```
Comment (webaaz) [inline tests/Unit/Endpoints/PaymentsTest.php:483]: missing phpdoc

File: src/Lib/StringUtils.php [hunk 7]
```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace Alma\API\Lib;
+
+class StringUtils
+{
+    /**
```
Comment (webaaz) [inline src/Lib/StringUtils.php:7]: missing comment on phpdoc

File: src/Lib/BoolUtils.php [hunk 8]
```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace Alma\API\Lib;
+
+class BoolUtils
+{
+    /**
```
Comment (webaaz) [inline src/Lib/BoolUtils.php:7]: missing description on phpdoc

File: src/Lib/BoolUtils.php [hunk 9]
```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace Alma\API\Lib;
+
+class BoolUtils
```
Comment (webaaz) [inline src/Lib/BoolUtils.php:5]: missing description on class

File: src/Endpoints/Payments.php [hunk 10]
```diff
@@ -355,4 +388,28 @@ public function sendSms($id)
         return true;
     }
 
+    /**
+     * @param $paymentId
```
Comment (webaaz) [inline src/Endpoints/Payments.php:392]: missing description

File: tests/Unit/Endpoints/PaymentsTest.php [hunk 11]
```diff
@@ -451,4 +452,152 @@ public function testFullRefundRequestError()
         $payments->fullRefund($id);
     }
 
+    /**
+     * @dataProvider addOrderStatusErrorPayloadProvider
```
Comment (webaaz) [inline tests/Unit/Endpoints/PaymentsTest.php:456]: missing description

File: tests/Unit/Endpoints/PaymentsTest.php [hunk 12]
```diff
@@ -451,4 +452,152 @@ public function testFullRefundRequestError()
...
+        $paymentEndpoint->addOrderStatusByMerchantOrderReference(
+            'payment_42',
+            'ref_3546',
+            'in progress',
+            false
+        );
+    }
+
+    /**
+     *
```
Comment (webaaz) [inline tests/Unit/Endpoints/PaymentsTest.php:514]: missing description

File: tests/Unit/Endpoints/PaymentsTest.php [hunk 13]
```diff
@@ -451,4 +452,152 @@ public function testFullRefundRequestError()
...
+
+        $paymentEndpoint->addOrderStatusByMerchantOrderReference(
+            $paymentId,
+            $merchantOrderReference,
+            $status,
+            $isShipped
+        );
+    }
+
+    public function AddOrderStatusProvider()
```
Comment (webaaz) [inline tests/Unit/Endpoints/PaymentsTest.php:558]: missing phpdoc

File: tests/Unit/Endpoints/PaymentsTest.php [hunk 14]
```diff
@@ -451,4 +452,152 @@ public function testFullRefundRequestError()
...
+            'With is shipped bool' => [
+                'paymentID' => 'payment_1234',
+                'merchantOrderReference' => 'merchant_order_123',
+                'status' => 'status_shipped',
+                'is_shipped' => true
+            ]
+        ];
+    }
+
+    public function addOrderStatusErrorPayloadProvider()
```
Comment (webaaz) [inline tests/Unit/Endpoints/PaymentsTest.php:575]: missing phpdoc

**Review by webaaz — CHANGES_REQUESTED** [review]

**Review by Benjamin-Freoua-Alma — APPROVED** [review]

**Review by joyet-simon — APPROVED** [review]

**Review by webaaz — APPROVED** [review]

**Pull request merged**